src/field_screen_effect.c

```diff
@@ -1274,3 +1279,119 @@ static void Task_EnableScriptAfterMusicFade(u8 taskId)
         ScriptContext_Enable();
     }
 }
+
+static const struct WindowTemplate sWindowTemplate_WhiteoutText =
+{
+    .bg = 0,
+    .tilemapLeft = 0,
+    .tilemapTop = 5,
+    .width = 30,
+    .height = 11,
+    .paletteNum = 15,
+    .baseBlock = 1,
+};
+
+static const u8 sWhiteoutTextColors[] = { TEXT_COLOR_TRANSPARENT, TEXT_COLOR_WHITE, TEXT_COLOR_DARK_GRAY };
+
+#define tState      data[0]
+#define tWindowId   data[1]
+#define tPrintState data[2]
+
+static bool8 PrintWhiteOutRecoveryMessage(u8 taskId, const u8 *text, u8 x, u8 y)
+{
+    u8 windowId = gTasks[taskId].tWindowId;
+
+    switch (gTasks[taskId].tPrintState)
+    {
+    case 0:
+        FillWindowPixelBuffer(windowId, PIXEL_FILL(0));
+        StringExpandPlaceholders(gStringVar4, text);
+        AddTextPrinterParameterized4(windowId, FONT_NORMAL, x, y, 1, 0, sWhiteoutTextColors, 1, gStringVar4);
+        gTextFlags.canABSpeedUpPrint = FALSE;
+        gTasks[taskId].tPrintState = 1;
+        break;
+    case 1:
+        RunTextPrinters();
+        if (!IsTextPrinterActive(windowId))
+        {
+            gTasks[taskId].tPrintState = 0;
+            return TRUE;
+        }
+        break;
+    }
+    return FALSE;
+}
+
+static void Task_RushInjuredPokemonToCenter(u8 taskId)
+{
+    u8 windowId;
+
+    switch (gTasks[taskId].tState)
+    {
+    case 0:
+        windowId = AddWindow(&sWindowTemplate_WhiteoutText);
+        gTasks[taskId].tWindowId = windowId;
+        Menu_LoadStdPalAt(BG_PLTT_ID(15));
+        FillWindowPixelBuffer(windowId, PIXEL_FILL(0));
+        PutWindowTilemap(windowId);
+        CopyWindowToVram(windowId, COPYWIN_FULL);
+
+        // Scene changes if last heal location was the player's house
+        if (IsLastHealLocation(HEAL_LOCATION_LITTLEROOT_TOWN_MAYS_HOUSE)
```

Comment:
I don't love this giant if statement. Is it possible to add them to a whitelist of "home" locations and then change `IsLastHealLocation` to `GetLastHealLocation` and check that against the list? ` IsRegularLandTrainer` is an example

Reply:
Okay now that I've looked at `IsLastHealLocation` I didn't realize you were checking 5 different conditions lmao, probably not possible